README
CLI-Startup

The README is used to introduce the module and provide instructions on
how to install the module, any machine dependencies it may have (for
example C compilers and installed libraries) and any other information
that should be provided before the module is installed.

A README file is required for CPAN modules since CPAN extracts the README
file from a module distribution so that people browsing the archive
can use it to get an idea of the module's uses. It is usually a good idea
to provide version information here so that people can decide whether
fixes for the module are worth downloading.


INSTALLATION

To install this module, run the following commands:

	perl Build.PL
	./Build
	./Build test
	./Build install

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc CLI::Startup

You can also look for information at:

    RT, CPAN's request tracker
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=CLI-Startup

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/CLI-Startup

    CPAN Ratings
        http://cpanratings.perl.org/d/CLI-Startup

    Search CPAN
        http://search.cpan.org/dist/CLI-Startup/


LICENSE AND COPYRIGHT

Copyright (C) 2011 Len Budney

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.



Changes
Revision history for CLI-Startup

0.21    2017-06-21
        Added perltidy to author tests. CLI::Startup.pm passes test.
0.20    2017-06-18
        Ensure that get_options()->{verbose} is always defined, even when it's 0.
        Consolidate the --verbose and -v options (-v is a magic pseudo-alias).
        Enable init() to work even if you only want the default options and no others.
        Add a verbose explanation of the default options to the man page.
0.19    2017-06-18
        Updated Perl dependency to 5.10.0 -- need regex (?&..) operator
        Removed case insensitivity of command-line options.
        Replaced deprecated JSON::Any with JSON::MaybeXS -- nice drop-in replacement.
        Cleaned up optspec validation code. Does the even-righter thing.
0.18    2017-06-17
        Cleaned up handling of option aliases to POSIX options will work.
        Added require_order option to improve POSIX compliance.
0.17    2017-03-24
        Removed test for unsupported optspec "foo:+".
        Added optional user message to die_usage().
        Changed --help and --manpage to exit 0 instead of 1.
        Updated author tests to work under Perl 5.24.
0.16    2014-10-17
        Removed alias --? for --help due to subtle bug.
        TODO: Add support for users to add aliases for --help.
0.15    2014-10-16
        Fixed tests that break when Config::Tiny isn't installed.
        Changed write_rcfile() to exit automatically.
0.14    2014-10-15
        Cleaned up some test code and fixed some failed tests.
0.13    2014-10-13
        Added explicit destructor, to turn off a Class:Std warning.
        Added documentation to the man page.
        Set explicit configs for Getopt::Long that are saner than the defaults.
        Added -? as an alias for --help.
0.12    2014-05-20
        Removed support for colon-based format for config files.
0.11    2014-05-19
        Added implicit dependencies missing from the Build.PL.
0.10    2014-05-18
        Improved config file processing to avoid corrupting non-config data in the file.
0.09    2014-05-17
        Switched from Config::Simple to Config::Any to add JSON, YAML, XML, and Perl support.
        Added --rcfile-format option to allow any of the above formats to be specified.
        Fixed a bug where some attributes were returned by reference instead of by copy.
        Bunches of cleanup to fix perlcritic warnings, some legitimate bugs among them.
        Moved POD to end of the file, making it easier to read.
0.08    2012-12-15
        Added example script: rsync wrapper similar to manpage example, but robuster.
0.07    2012-09-23
        Put back missing sort of command-line options in help message. TNX Tim Heaney.
0.06    2012-04-07
        Fixed a small typo in the POD example code. TNX Tim Heaney.
0.05    2011-08-20
        Added workaround for instability in File::HomeDir. Grumble. See:
        http://www.dagolden.com/index.php/1238/upgrading-filehomedir-might-lose-your-config-files/
0.04    2011-07-10
        Replaced $HOME with platform-independent File::HomeDir.
0.03    2011-07-09
        Fix documentation and one failed test.
0.02    2011-07-08
        First feature-complete version.
0.01    2011-07-04
        Initial revision


Makefile.PL
# Note: this file was auto-generated by Module::Build::Compat version 0.4220
require 5.010000;
use ExtUtils::MakeMaker;
WriteMakefile
(
  'EXE_FILES' => [],
  'NAME' => 'CLI::Startup',
  'VERSION_FROM' => 'lib/CLI/Startup.pm',
  'PL_FILES' => {},
  'INSTALLDIRS' => 'site',
  'PREREQ_PM' => {
                   'Test::Trap' => 0,
                   'Class::Std' => '0.01',
                   'Hash::Merge' => 0,
                   'File::Basename' => 0,
                   'Exporter' => 0,
                   'Data::Dumper' => 0,
                   'Cwd' => 0,
                   'Pod::Text' => '3.01',
                   'Text::CSV' => 0,
                   'English' => 0,
                   'Test::More' => 0,
                   'Config::Any' => 0,
                   'List::Util' => 0,
                   'strict' => 0,
                   'Test::Exception' => 0,
                   'Getopt::Long' => '2.38',
                   'Clone' => 0,
                   'warnings' => 0,
                   'File::HomeDir' => '0.97',
                   'base' => 0,
                   'File::Copy' => 0,
                   'Symbol' => 0,
                   'Readonly' => 0
                 }
)
;


Build.PL
use strict;
use warnings;

use Module::Build;

my $class = Module::Build->subclass( code => <<'EOF' );

    sub ACTION_authortest {
        my ($self) = @_;

        $self->depends_on('build');
        $self->depends_on('manifest');
        $self->depends_on('distmeta');

        $self->test_files( qw< t xt/ > );
        $self->recursive_test_files(1);

        $self->depends_on('test');

        return;
    }

    sub ACTION_distdir {
        my ($self) = @_;

        $self->depends_on('authortest');

        return $self->SUPER::ACTION_distdir();
    }
EOF

my $builder = $class->new(
    module_name         => 'CLI::Startup',
    license             => 'perl',
    dist_author         => q{Len Budney <len.budney@gmail.com>},
    dist_version_from   => 'lib/CLI/Startup.pm',

    requires => {
        'perl'            => '5.10.0',
        'Class::Std'      => '0.01',
        'Clone'           => 0,
        'Config::Any'     => 0,
        'Data::Dumper'    => 0,
        'English'         => 0,
        'Exporter'        => 0,
        'File::Basename'  => 0,
        'File::Copy'      => 0,
        'File::HomeDir'   => '0.97',
        'Getopt::Long'    => '2.38',
        'Hash::Merge'     => 0,
        'List::Util'      => 0,
        'Pod::Text'       => '3.01',
        'Symbol'          => 0,
        'Text::CSV'       => 0,
    },

    recommends => {
        'Config::Tiny'        => 0, # To read INI configs
        'Config::INI::Writer' => 0, # To write INI configs
        'JSON::MaybeXS'       => 0, # To write JSON configs
        'YAML::Any'           => 0, # To write YAML configs
        'XML::Simple'         => 0, # To write XML configs
    },

    build_requires => {
        'Cwd'             => 0,
        'Test::More'      => 0,
        'Test::Trap'      => 0,
        'Test::Exception' => 0,

        # Shut up Test::Prereq
        'base'            => 0,
        'strict'          => 0,
        'warnings'        => 0,
    },

    configure_requires => {
        'Module::Build' => 0.33,
    },

    add_to_cleanup      => [qw{
        *.bak
        CLI-Startup-*
    }],
    create_makefile_pl => 'traditional',
);

$builder->create_build_script();
